go/consensus/tendermint/apps/roothash/transactions.go

```diff
@@ -90,7 +118,7 @@ func (app *rootHashApplication) getRuntimeState(
 			"round", rtState.CurrentBlock.Header.Round,
 		)
 
-		rtState.Round.Transition(rtState.CurrentBlock)
+		rtState.Round.Transition(rtState.CurrentBlock, ctx.BlockHeight())
```

Comment:
With the `Round` now recording the block height at which the given round was finalized, we should probably not trigger the transition here? It should instead be triggered when the roothash block is finalized (e.g., in `tryFinalizeExecutor` and `emitEmptyBlock`). Otherwise the `CurrentBlockHeight` will be wrong and will not be updated until the first commit is received for the next round.

Reply:
fixed! Note did also have to move the `Round.Reset()` out of `round.Transition()` since in `tryFinalizeExecutor` we also set `.Finalized = true`  and i don't think we want to reset that until we're in `postProcessFinalizedBlock`